.travis.yml

```diff
@@ -24,18 +30,20 @@ matrix:
       env: SYMFONY_VERSION=2.8.*
     - php: 5.6
       env: SYMFONY_VERSION=3.0.*
-
-sudo: false
+    - php: 7.2
+      env: SYMFONY_VERSION=4.0.*
 
 cache:
   directories:
-    - $HOME/.composer/cache/files
+    - $HOME/.composer
 
 before_install:
   - composer self-update
   - if [ "$SYMFONY_VERSION" != "" ]; then composer require --dev --no-update symfony/symfony=$SYMFONY_VERSION; fi
   - INI_FILE=~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini;
   - echo memory_limit = -1 >> $INI_FILE
+  # turn off XDebug
+  - phpenv config-rm xdebug.ini || return 0
```

Comment:
Could you explain the `|| return 0` bit?  Is that so that this doesn't give a non-zero return code and fail the build?  If so, shouldn't that be `&& return 0`?  Sorry if I'm missing something here!

Reply:
I'm going to ping @keradus here, author of
https://github.com/FriendsOfPHP/PHP-CS-Fixer/commit/5759ef8db7dcb356f2db6eb45310dcd860486c69
as I copied this line to the Twig repo as well I start to wonder how this works as well :D

Reply:
this is bash.
`commandA && commandB` means execute `commandA`, and only if it was successful execute `commandB`
`commandA || commandB` means execute `commandA`, and only if it was failure execute `commandB`
overall returned status in both cases is status of last executed command

What I believe confused you is that "was successful" actually means "exit status/return status === 0".
It's bash arithmetic. you cannot substitute the commands with exit status they are suppose to return, like here in case of error - `1 || 0` - and then use boolean logic on that, at you would get overall outcome of `1`, while in fact exit status of alternation would be `0`

---------------

so, here:
- if `phpenv config-rm xdebug.ini` was successful (exit status = 0), continue to next command of `before_install`
- if `phpenv config-rm xdebug.ini` was a failure (exit status != 0), then override exit status to 0 (by `return 0`), and then continue to next command of `before_install`.

What does it means for us? if disabling XDebug failed, we don't care and still proceed with Travis job.
Where can it fail? it will fail for sure for old hhvm, that's why I originally added that alternation.
Does it fail with new hhvm version? Does if fail with PHP 7, which doesnt have xdebug by default? I have no idea.
I needed the rescue of failure on hhvm, that was the way ;)

---------------

hope I helped.

Reply:
thanks so much man :+1:  great explanation : D
so much stuff about bash I still have to learn

Reply:
Yes, that's a great explanation.  Thanks